.github/workflows/build.yml

```
@@ -98,3 +98,38 @@ jobs:
       - name: Setup tmate debug session
         uses: mxschmitt/action-tmate@v3
         if: env.RUN_TMATE
+  test:
+    name: Run Go Package Tests
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Download Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: '^1.20.2'
+        id: go
+
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v3
+
+      - name: "Retrieve package location"
+        run: |
+            echo PACKAGE_LOCATION="$(find ./ -name 'test')/../" >> $GITHUB_OUTPUT
+        id: packagelocation
+
+      - name: Generate Mocks and Run Tests
+        working-directory: ${{ steps.packagelocation.outputs.PACKAGE_LOCATION }}
+        run: |
+          export GOPATH="$HOME/go/"
+          export PATH=$PATH:$GOPATH/bin
+          go install github.com/golang/mock/mockgen@v1.6.0
+          go install github.com/jstemmer/go-junit-report@latest
```

Replace with a fixed version:
```suggestion
          go install github.com/jstemmer/go-junit-report@2.0.0
```